src/components/FolderAndLink.tsx

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { Link1, getFolders, getLinksByFolderID } from "../api";
+import FolderList from "./FolderList/FolderList";
+
+import useAsync from "../Hooks/useAsync";
+
+import styled from "styled-components";
+import { device } from "../style/device";
+import CardList from "./CardList";
+
+function FolderAndLink() {
+  const [selectedFolderId, setSelectedFolderId] = useState<number>();
+
+  const DEFAULT_FOLDER = 1;
+  // 폴더
+
+  const { data: folderData } = useAsync(() => getFolders(DEFAULT_FOLDER));
+
+  const getLinks = useCallback(() => {
+    return getLinksByFolderID(DEFAULT_FOLDER, selectedFolderId);
+  }, [selectedFolderId]);
+
+  const { data: linkData } = useAsync(getLinks);
+
+  const setFolderLink = (folder_id: number) => {
```

Comment:
```suggestion
  const setFolderLink = (folder_id: number | undefined) => {
```
FolderList에서 setFolderLink의 함수 타입이랑 맞춰주기 위해서 타입 수정이 필요해보입니다.